lib/react_client.dart

```diff
@@ -286,19 +337,7 @@ ReactComponentFactory _registerComponent(ComponentFactory componentFactory, [Ite
  * create dart-react registered component for html tag.
  */
 _reactDom(String name) {
-  var call = (Map props, [dynamic children]) {
-    _convertBoundValues(props);
-    _convertEventHandlers(props);
-    if (props.containsKey('style')) {
-      props['style'] = new JsObject.jsify(props['style']);
-    }
-    if (children is Iterable) {
-      children = new JsArray.from(children);
-    }
-    return _React['createElement'].apply([name, newJsMap(props), children]);
-  };
-
-  return new ReactComponentFactoryProxy(_React['DOM'][name], call);
+  return new ReactDomComponentFactoryProxy(name);
```

Comment:
Cool. We'll have to update the `react_test_utils` wrapper accordingly, though, so typing continues to work with DOM components:

`react_test_utils.dart`:
```dart
/// Returns the 'type' of a component.
///
/// For a DOM components, this with return the String corresponding to its tagName ('div', 'a', etc.).
/// For React.createClass()-based components, this with return the React class as a JsFunction.
dynamic getComponentType(ReactComponentFactory componentFactory) {
  if (componentFactory is ReactComponentFactoryProxy) {
    return componentFactory.reactComponentFactory['type'];
  }
  if (componentFactory is ReactDomComponentFactoryProxy) {
    return componentFactory.name;
  }
  return null;
}
```